static/js/src/support-chart.js

```diff
@@ -0,0 +1,550 @@
+ * @param {Obj} margin
+ *
+ * Find the postion of the Y-axis ticks and adds a horizontal
+ * line above each label and one that spans the chart area
+ */
+function addYAxisHorizontalLines(svg, yAxis, width, margin) {
+  const tickPositions = getTickPositions(yAxis);
+
+  const lineAdjustment = 27;
+
+  tickPositions.forEach((posY, index) => {
+    addHorizontalLine(svg, -margin.left, 0, posY - lineAdjustment, "#D9D9D9", 2);
+
+    addHorizontalLine(svg, 0, width + margin.right, posY - lineAdjustment, "#D9D9D9", index === 0 ? 2 : 1);
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * @param {*} versionAxis
+ *
+ * Add version axis to chart
+ */
+function addVersionAxis(svg, versionAxis) {
+  svg.append("g").attr("class", "version axis").call(versionAxis);
+}
+
+/**
+ *
+ * @param {*} svg
+ *
+ * Clean up unwanted elements on chart put in by d3.js
+ */
+function cleanUpChart(svg) {
+  svg.selectAll(".domain").remove();
+}
+
+/**
+ *
+ * @param {*} svg
+ *
+ * Embolden LTS labels on y axis**
+ *
+ */
+function emboldenLTSLabels(svg) {
+  svg.selectAll(".tick text tspan").select(function () {
+    var text = this.textContent;
+    console.log("test", text);
+    if (text.includes("LTS")) {
+      this.classList.add("chart__label--bold");
+    }
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * @param {String} highlightVersion
+ * 
+ * Set version axis labels
+ */
+function highlightChartRow(svg, highlightVersion) {
+  svg.selectAll(".tick text").select(function () {
+    var text = this.textContent;
+    var isNotHighlightedVersion =
+      highlightVersion && !text.includes(highlightVersion);
+    var isYearLabel = text.includes("20") && !text.includes("Ubuntu ");
+
+    if (isNotHighlightedVersion) {
+      this.classList.add("chart__label--transparent");
+    }
+
+    if (isYearLabel) {
+      this.classList.remove("chart__label--transparent");
+    }
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * 
+ * Set version axis labels
+ */
+function setVersionAxisLabels(svg, taskVersions) {
+  svg.selectAll(".version .tick text").select(function (tickLabel, index) {
+    this.textContent = taskVersions[index];
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * @param {Int} height
+ *
+ * Adds vertical lines to the x axis
+ */
+function addXAxisVerticalLines(svg, height, margin) {
+  svg
+    .selectAll(".x.axis .tick line")
+    .attr("y1", height - margin.bottom - margin.top)
+    .attr("y2", 26);
+}
+
+/**
+ *
+ * @param {String} chartSelector
+ * @param {Object} taskStatus
+ *
+ * Builds key for supplied chart based on task status
+ */
+function buildChartKey(chartSelector, taskStatus) {
+  var taskStatusKeys = Object.keys(taskStatus);
+  var rectDimensions = 25;
+  var rowHeight = rectDimensions + 5;
+  var gapBetweenRectAndText = 10;
+  var verticalTextAlignmentOffset = 16;
+
+  var chartKey = d3
+    .select(chartSelector)
+    .append("svg")
+    .attr("class", "chart-key")
+    .attr("width", "550")
+    .attr("height", rowHeight * taskStatusKeys.length);
```

Comment:
The hardcoded width is causing issues on medium and small screens
In some cases, it gets cut and parts of it, doesnt show
and in other cases it causes weird screen issues

Example 1:
![image](https://github.com/canonical/ubuntu.com/assets/31866582/5f7a1c79-5d73-430c-b1e6-4a991cb7cee1)
![image](https://github.com/canonical/ubuntu.com/assets/31866582/570e2d48-02fa-4741-ba7a-543a3f68026b)


Example 2:
![image](https://github.com/canonical/ubuntu.com/assets/31866582/fd2842a3-875a-46b2-8ce5-18a0eb49ef5c)
![image](https://github.com/canonical/ubuntu.com/assets/31866582/e0c23b86-4a8c-461c-af14-afce915fad98)